File: ros_ws/src/my_robot_tutorials/scripts/battery_monitor.py
#!/usr/bin/env python

import rospy
from my_robot_msgs.srv import SetLedPanel


def main(args=None):
    rospy.init_node("battery_monitor")
    
    client = rospy.ServiceProxy("set_led", SetLedPanel)
    rate = rospy.Rate(1.0 / 7.0)
    battery_state = False

    rospy.loginfo("battery_monitor node has been created")

    while not rospy.is_shutdown():
        rospy.wait_for_service("set_led")

        try:            
            response = client(3, battery_state)

            if response.success:
                rospy.loginfo("Led changed successfully")
            else:
                rospy.loginfo("Failed to change led")

        except rospy.ServiceException as e:
            rospy.logwarn("Service call failed")

        battery_state = not battery_state
        
        rate.sleep()


if __name__ == "__main__":
    main()

File: ros_ws/src/my_robot_tutorials/src/battery_monitor.cpp
#include <ros/ros.h>
#include <my_robot_msgs/SetLedPanel.h>

int main(int argc, char **argv)
{
    ros::init(argc, argv, "battery_monitor");
    ros::NodeHandle nh;
    ros::ServiceClient client = nh.serviceClient<my_robot_msgs::SetLedPanel>("set_led");
    ros::Rate rate(1.0 / 7.0);
    bool batteryState = false;

    ROS_INFO("battery_monitor has been created");

    while(ros::ok()) {
        my_robot_msgs::SetLedPanel srv;
        srv.request.led_number = 3;
        srv.request.led_state = batteryState;

        if(client.call(srv)) {
            ROS_INFO("Led state change successfully: %d", (int)srv.response.success);
        }
        else {
            ROS_WARN("Service call failed");
        }
        batteryState = !batteryState;
        rate.sleep();
    }

    return 0;
}